### PR TITLE
Add support for 256-bit tags

### PR DESCRIPTION
**Pull request by jedisct1**
Update the test vectors and implementations accordingly.

Once we have a proof that AEGIS is context committing, we can expand on when to use 128 vs 256 bit tags in the context of commitment.

### CODE REVIEW

File: draft-irtf-cfrg-aegis-aead.md [hunk 1]
```diff
@@ -193,8 +193,8 @@ This document is a product of the Crypto Forum Research Group (CFRG). It is not
 
 This document describes the AEGIS-128L and AEGIS-256 authenticated encryption with associated data (AEAD) algorithms {{AEGIS}}, which were chosen as additional finalists for high-performance applications in the Competition for Authenticated Encryption: Security, Applicability, and Robustness (CAESAR). Whilst AEGIS-128 was selected as a winner for this use case, AEGIS-128L has a better security margin alongside improved performance and AEGIS-256 uses a 256-bit key {{LIMS21}}. All variants of AEGIS are constructed from the AES encryption round function {{!FIPS-AES=FIPS.197.2001}}. This document specifies:
 
-- AEGIS-128L, which has a 128-bit key, a 128-bit nonce, a 1024-bit state, a 128-bit authentication tag, and processes 256-bit input blocks.
-- AEGIS-256, which has a 256-bit key, a 256-bit nonce, a 768-bit state, a 128-bit authentication tag, and processes 128-bit input blocks.
+- AEGIS-128L, which has a 128-bit key, a 128-bit nonce, a 1024-bit state, a 128-bit or 256-bit authentication tag, and processes 256-bit input blocks.
+- AEGIS-256, which has a 256-bit key, a 256-bit nonce, a 768-bit state, a 128-bit or 256-bit authentication tag, and processes 128-bit input blocks.
```
Comment (samuel-lucas6) [inline draft-irtf-cfrg-aegis-aead.md:197]: What are your thoughts on '128- or 256-bit'? It feels more consistent with '(128 or 256 bits)' to me.

File: draft-irtf-cfrg-aegis-aead.md [hunk 2]
```diff
@@ -567,7 +567,10 @@ t = S2 ^ (LE64(ad_len) || LE64(msg_len))
 
 Repeat(7, Update(t, t))
 
-tag = S0 ^ S1 ^ S2 ^ S3 ^ S4 ^ S5 ^ S6
+If tag_length == 16: # 128 bits
```
Comment (samuel-lucas6) [inline draft-irtf-cfrg-aegis-aead.md:570]: The current pseudocode uses lowercase `if` and `else` (e.g. [here](https://jedisct1.github.io/draft-aegis-aead/draft-irtf-cfrg-aegis-aead.html#section-3.2-9)).

File: draft-irtf-cfrg-aegis-aead.md [hunk 3]
```diff
@@ -891,7 +897,9 @@ combined_ct = ct || tag
 
 # Security Considerations
 
-AEGIS-256 offers 256-bit message security against plaintext and state recovery, whereas AEGIS-128L offers 128-bit security. Both have a 128-bit authentication tag, which implies that a given tag may verify under multiple keys. However, assuming AEGIS is key-committing, finding equivalent keys is expected to be significantly more difficult than for authentication schemes based on polynomial evaluation, such as GCM and Poly1305.
+AEGIS-256 offers 256-bit message security against plaintext and state recovery, whereas AEGIS-128L offers 128-bit security.
+
+An authentication tag may verify under multiple keys. Assuming AEGIS is key-committing, finding equivalent keys is expected to require ~2<sup>64<sup> attempts with a 128-bit authentication tag, and ~2<sup>128</sup> attempts with a 256-bit tag.
```
Comment (samuel-lucas6) [inline draft-irtf-cfrg-aegis-aead.md:902]: Missing `</sup>` with the `~2<sup>64<sup>`. Last comma before the 'and' can also be removed.

**Review by samuel-lucas6 — CHANGES_REQUESTED** [review]
Nice work! A few style things and good to go.

File: draft-irtf-cfrg-aegis-aead.md (hunk 1 above)
Comment (jedisct1) [reply draft-irtf-cfrg-aegis-aead.md:197]: Indeed, `128- or 256-bit` looks better. Changed, thanks!

File: draft-irtf-cfrg-aegis-aead.md (hunk 3 above)
Comment (jedisct1) [reply draft-irtf-cfrg-aegis-aead.md:902]: Done.

**Pull request merged**